src/vm/threads.cpp

```diff
@@ -6551,6 +6559,106 @@ Thread *ThreadStore::GetThreadList(Thread *cursor)
     return GetAllThreadList(cursor, (Thread::TS_Unstarted | Thread::TS_Dead), 0);
 }
 
+const LONG ThreadStore::s_DeadThreadCountThresholdForGCTrigger = 25;
+const DWORD ThreadStore::s_DeadThreadGCTriggerPeriodMilliseconds = 1000 * 60 * 10;
+
+//
+// Dead thread count for GC trigger
+// - Interlocked operations are used, as a background GC thread may also be involved in reading/writing the value
+// - The thread store lock could theoreticaly be used, but it would add new GC suspension points and it is unclear that this is
+//   ok at these points
+//
+
+LONG ThreadStore::GetDeadThreadCountForGCTrigger()
+{
+    // Usage of this value should not assume a nonnegative value, see Increment and Decrement
+    return FastInterlockCompareExchange(&m_DeadThreadCountForGCTrigger, -1, -1);
```

Comment:
Why not just `return m_DeadThreadCountForGCTrigger`?  That would also be atomic.  Does this need to be a full memory barrier as well?